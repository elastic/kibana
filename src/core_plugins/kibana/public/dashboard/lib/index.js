export { saveDashboard } from './save_dashboard';
export { getAppStateDefaults } from './get_app_state_defaults';
export { migrateAppState } from './migrate_app_state';
