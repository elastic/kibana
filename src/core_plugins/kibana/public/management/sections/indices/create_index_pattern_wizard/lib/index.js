export { canAppendWildcard } from './can_append_wildcard';

export { createReasonableWait } from './create_reasonable_wait';

export { getIndices } from './get_indices';

export { getMatchedIndices } from './get_matched_indices';

export { containsInvalidCharacters } from './contains_invalid_characters';

export { extractTimeFields } from './extract_time_fields';
