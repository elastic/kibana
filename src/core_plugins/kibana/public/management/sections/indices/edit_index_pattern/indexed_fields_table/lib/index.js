export { getFieldFormat } from './get_field_format';
