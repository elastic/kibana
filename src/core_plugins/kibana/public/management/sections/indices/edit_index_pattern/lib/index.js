export { booleanTemplate } from './boolean_template';
export { getFieldFormat } from './get_field_format';
export { fieldWildcardProvider } from './field_wildcard';
