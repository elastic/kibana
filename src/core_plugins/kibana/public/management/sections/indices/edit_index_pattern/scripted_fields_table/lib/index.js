export { getItemsOnPage } from './get_items_on_page';
