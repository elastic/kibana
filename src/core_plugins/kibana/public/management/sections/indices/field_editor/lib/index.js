export { copyField } from './copy_field';
export { getDefaultFormat } from './get_default_format';
