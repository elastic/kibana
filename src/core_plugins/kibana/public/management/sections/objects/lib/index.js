export * from './get_in_app_url';
export * from './get_relationships';
export * from './get_saved_object_counts';
export * from './get_saved_object_icon';
export * from './get_saved_object_label';
export * from './import_file';
export * from './parse_query';
export * from './resolve_saved_objects';
export * from './retrieve_and_export_docs';
export * from './save_to_file';
export * from './scan_all_types';
