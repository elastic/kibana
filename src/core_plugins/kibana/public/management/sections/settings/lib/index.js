export { isDefaultValue } from './is_default_value';
export { toEditableConfig } from './to_editable_config';
export { getCategoryName } from './get_category_name';
export { DEFAULT_CATEGORY } from './default_category';
export { getAriaName } from './get_aria_name';
