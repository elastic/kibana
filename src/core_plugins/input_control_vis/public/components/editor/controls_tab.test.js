/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import sinon from 'sinon';
import { shallowWithIntl, mountWithIntl } from 'test_utils/enzyme_helpers';
import { findTestSubject } from '@elastic/eui/lib/test';
import { getIndexPatternMock } from './__tests__/get_index_pattern_mock';
import {
  ControlsTab,
} from './controls_tab';

const indexPatternsMock = {
  get: getIndexPatternMock
};
const scopeMock = {
  vis: {
    API: {
      indexPatterns: indexPatternsMock
    },
  },
  editorState: {
    params: {
      'controls': [
        {
          'id': '1',
          'indexPattern': 'indexPattern1',
          'fieldName': 'keywordField',
          'label': 'custom label',
          'type': 'list',
          'options': {
            'type': 'terms',
            'multiselect': true,
            'size': 5,
            'order': 'desc'
          }
        },
        {
          'id': '2',
          'indexPattern': 'indexPattern1',
          'fieldName': 'numberField',
          'label': '',
          'type': 'range',
          'options': {
            'step': 1
          }
        }
      ]
    }
  }
};
let stageEditorParams;

beforeEach(() => {
  stageEditorParams = sinon.spy();
});

test('renders ControlsTab', () => {
  const component = shallowWithIntl(<ControlsTab.WrappedComponent
    scope={scopeMock}
    editorState={scopeMock.editorState}
    stageEditorParams={stageEditorParams}
  />);
  expect(component).toMatchSnapshot(); // eslint-disable-line
});

describe('behavior', () => {

  test('add control button', () => {
    const component = mountWithIntl(<ControlsTab.WrappedComponent
      scope={scopeMock}
      editorState={scopeMock.editorState}
      stageEditorParams={stageEditorParams}
    />);
    findTestSubject(component, 'inputControlEditorAddBtn').simulate('click');
    // Use custom match function since control.id is dynamically generated and never the same.
    sinon.assert.calledWith(stageEditorParams, sinon.match((newParams) => {
      if (newParams.controls.length !== 3) {
        return false;
      }
      return true;
    }, 'control not added to editorState.params'));
  });

  test('remove control button', () => {
    const component = mountWithIntl(<ControlsTab.WrappedComponent
      scope={scopeMock}
      editorState={scopeMock.editorState}
      stageEditorParams={stageEditorParams}
    />);
    findTestSubject(component, 'inputControlEditorRemoveControl0').simulate('click');
    const expectedParams = {
      'controls': [
        {
          'id': '2',
          'indexPattern': 'indexPattern1',
          'fieldName': 'numberField',
          'label': '',
          'type': 'range',
          'options': {
            'step': 1
          }
        }
      ]
    };
    sinon.assert.calledWith(stageEditorParams, sinon.match(expectedParams));
  });


  test('move down control button', () => {
    const component = mountWithIntl(<ControlsTab.WrappedComponent
      scope={scopeMock}
      editorState={scopeMock.editorState}
      stageEditorParams={stageEditorParams}
    />);
    findTestSubject(component, 'inputControlEditorMoveDownControl0').simulate('click');
    const expectedParams = {
      'controls': [
        {
          'id': '2',
          'indexPattern': 'indexPattern1',
          'fieldName': 'numberField',
          'label': '',
          'type': 'range',
          'options': {
            'step': 1
          }
        },
        {
          'id': '1',
          'indexPattern': 'indexPattern1',
          'fieldName': 'keywordField',
          'label': 'custom label',
          'type': 'list',
          'options': {
            'type': 'terms',
            'multiselect': true,
            'size': 5,
            'order': 'desc'
          }
        }
      ]
    };
    sinon.assert.calledWith(stageEditorParams, sinon.match(expectedParams));
  });

  test('move up control button', () => {
    const component = mountWithIntl(<ControlsTab.WrappedComponent
      scope={scopeMock}
      editorState={scopeMock.editorState}
      stageEditorParams={stageEditorParams}
    />);
    findTestSubject(component, 'inputControlEditorMoveUpControl1').simulate('click');
    const expectedParams = {
      'controls': [
        {
          'id': '2',
          'indexPattern': 'indexPattern1',
          'fieldName': 'numberField',
          'label': '',
          'type': 'range',
          'options': {
            'step': 1
          }
        },
        {
          'id': '1',
          'indexPattern': 'indexPattern1',
          'fieldName': 'keywordField',
          'label': 'custom label',
          'type': 'list',
          'options': {
            'type': 'terms',
            'multiselect': true,
            'size': 5,
            'order': 'desc'
          }
        }
      ]
    };
    sinon.assert.calledWith(stageEditorParams, sinon.match(expectedParams));
  });
});
