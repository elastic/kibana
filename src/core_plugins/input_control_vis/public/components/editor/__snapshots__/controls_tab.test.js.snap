// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ControlsTab 1`] = `
<div>
  <ControlEditor
    controlIndex={0}
    controlParams={
      Object {
        "fieldName": "keywordField",
        "id": "1",
        "indexPattern": "indexPattern1",
        "label": "custom label",
        "options": Object {
          "multiselect": true,
          "order": "desc",
          "size": 5,
          "type": "terms",
        },
        "type": "list",
      }
    }
    getIndexPattern={[Function]}
    getIndexPatterns={[Function]}
    handleCheckboxOptionChange={[Function]}
    handleFieldNameChange={[Function]}
    handleIndexPatternChange={[Function]}
    handleLabelChange={[Function]}
    handleNumberOptionChange={[Function]}
    handleRemoveControl={[Function]}
    key="1"
    moveControl={[Function]}
  />
  <ControlEditor
    controlIndex={1}
    controlParams={
      Object {
        "fieldName": "numberField",
        "id": "2",
        "indexPattern": "indexPattern1",
        "label": "",
        "options": Object {
          "step": 1,
        },
        "type": "range",
      }
    }
    getIndexPattern={[Function]}
    getIndexPatterns={[Function]}
    handleCheckboxOptionChange={[Function]}
    handleFieldNameChange={[Function]}
    handleIndexPatternChange={[Function]}
    handleLabelChange={[Function]}
    handleNumberOptionChange={[Function]}
    handleRemoveControl={[Function]}
    key="2"
    moveControl={[Function]}
  />
  <div
    className="kuiSideBarFormRow"
  >
    <div
      className="kuiSideBarFormRow__control kuiFieldGroupSection--wide"
    >
      <select
        aria-label="Select control type"
        className="kuiSelect"
        onChange={[Function]}
        value="list"
      >
        <option
          value="range"
        >
          Range slider
        </option>
        <option
          value="list"
        >
          Options list
        </option>
      </select>
    </div>
    <KuiButton
      buttonType="primary"
      data-test-subj="inputControlEditorAddBtn"
      icon={
        <KuiButtonIcon
          type="create"
        />
      }
      onClick={[Function]}
      type="button"
    >
      Add
    </KuiButton>
  </div>
</div>
`;
